ktor-network/jvm/src/io/ktor/network/util/Utils.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.network.util
+
+import io.ktor.network.sockets.*
+import kotlinx.coroutines.*
+
+/**
+ * Wrap [block] into [withTimeout] wrapper and throws [SocketTimeoutException] if timeout exceeded.
+ */
+suspend fun CoroutineScope.withSocketTimeout(socketTimeout: Long, block: suspend CoroutineScope.() -> Unit) {
```

Comment:
Probably we shouldn't expose this method.

Reply:
Good point, I made it `internal`.